{
	"document": {
		"aggregate_severity": {
			"namespace": "https://access.redhat.com/security/updates/classification/",
			"text": "Important"
		},
		"category": "csaf_security_advisory",
		"csaf_version": "2.0",
		"distribution": {
			"text": "Copyright Â© Red Hat, Inc. All rights reserved.",
			"tlp": {
				"label": "WHITE",
				"url": "https://www.first.org/tlp/"
			}
		},
		"lang": "en",
		"notes": [
			{
				"category": "summary",
				"text": "An update for python-setuptools is now available for Red Hat Enterprise Linux 9.\n\nRed Hat Product Security has rated this update as having a security impact of Important. A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE link(s) in the References section.",
				"title": "Topic"
			},
			{
				"category": "general",
				"text": "The python-setuptools package provides a collection of enhancements to Python distribution utilities allowing convenient building and distribution of Python packages.\n\nSecurity Fix(es):\n\n* pypa/setuptools: Remote code execution via download functions in the package_index module in pypa/setuptools (CVE-2024-6345)\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.",
				"title": "Details"
			},
			{
				"category": "legal_disclaimer",
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use"
			}
		],
		"publisher": {
			"category": "vendor",
			"contact_details": "https://access.redhat.com/security/team/contact/",
			"issuing_authority": "Red Hat Product Security is responsible for vulnerability handling across all Red Hat products and services.",
			"name": "Red Hat Product Security",
			"namespace": "https://www.redhat.com"
		},
		"references": [
			{
				"category": "self",
				"summary": "https://access.redhat.com/errata/RHSA-2024:5534",
				"url": "https://access.redhat.com/errata/RHSA-2024:5534"
			},
			{
				"category": "external",
				"summary": "https://access.redhat.com/security/updates/classification/#important",
				"url": "https://access.redhat.com/security/updates/classification/#important"
			},
			{
				"category": "external",
				"summary": "2297771",
				"url": "https://bugzilla.redhat.com/show_bug.cgi?id=2297771"
			},
			{
				"category": "self",
				"summary": "Canonical URL",
				"url": "https://security.access.redhat.com/data/csaf/v2/advisories/2024/rhsa-2024_5534.json"
			}
		],
		"title": "Red Hat Security Advisory: python-setuptools security update",
		"tracking": {
			"current_release_date": "2024-12-30T14:44:06+00:00",
			"generator": {
				"date": "2024-12-30T14:44:06+00:00",
				"engine": {
					"name": "Red Hat SDEngine",
					"version": "4.2.4"
				}
			},
			"id": "RHSA-2024:5534",
			"initial_release_date": "2024-08-19T02:36:42+00:00",
			"revision_history": [
				{
					"date": "2024-08-19T02:36:42+00:00",
					"number": "1",
					"summary": "Initial version"
				},
				{
					"date": "2024-08-19T02:36:42+00:00",
					"number": "2",
					"summary": "Last updated version"
				},
				{
					"date": "2024-12-30T14:44:06+00:00",
					"number": "3",
					"summary": "Last generated version"
				}
			],
			"status": "final",
			"version": "3"
		}
	},
	"product_tree": {
		"branches": [
			{
				"branches": [
					{
						"branches": [
							{
								"category": "product_name",
								"name": "Red Hat Enterprise Linux BaseOS (v. 9)",
								"product": {
									"name": "Red Hat Enterprise Linux BaseOS (v. 9)",
									"product_id": "BaseOS-9.4.0.Z.MAIN.EUS",
									"product_identification_helper": {
										"cpe": "cpe:/o:redhat:enterprise_linux:9::baseos"
									}
								}
							}
						],
						"category": "product_family",
						"name": "Red Hat Enterprise Linux"
					},
					{
						"branches": [
							{
								"category": "product_version",
								"name": "python-setuptools-0:53.0.0-12.el9_4.1.src",
								"product": {
									"name": "python-setuptools-0:53.0.0-12.el9_4.1.src",
									"product_id": "python-setuptools-0:53.0.0-12.el9_4.1.src",
									"product_identification_helper": {
										"purl": "pkg:rpm/redhat/python-setuptools@53.0.0-12.el9_4.1?arch=src"
									}
								}
							}
						],
						"category": "architecture",
						"name": "src"
					},
					{
						"branches": [
							{
								"category": "product_version",
								"name": "python3-setuptools-0:53.0.0-12.el9_4.1.noarch",
								"product": {
									"name": "python3-setuptools-0:53.0.0-12.el9_4.1.noarch",
									"product_id": "python3-setuptools-0:53.0.0-12.el9_4.1.noarch",
									"product_identification_helper": {
										"purl": "pkg:rpm/redhat/python3-setuptools@53.0.0-12.el9_4.1?arch=noarch"
									}
								}
							},
							{
								"category": "product_version",
								"name": "python3-setuptools-wheel-0:53.0.0-12.el9_4.1.noarch",
								"product": {
									"name": "python3-setuptools-wheel-0:53.0.0-12.el9_4.1.noarch",
									"product_id": "python3-setuptools-wheel-0:53.0.0-12.el9_4.1.noarch",
									"product_identification_helper": {
										"purl": "pkg:rpm/redhat/python3-setuptools-wheel@53.0.0-12.el9_4.1?arch=noarch"
									}
								}
							}
						],
						"category": "architecture",
						"name": "noarch"
					}
				],
				"category": "vendor",
				"name": "Red Hat"
			}
		],
		"relationships": [
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python-setuptools-0:53.0.0-12.el9_4.1.src as a component of Red Hat Enterprise Linux BaseOS (v. 9)",
					"product_id": "BaseOS-9.4.0.Z.MAIN.EUS:python-setuptools-0:53.0.0-12.el9_4.1.src"
				},
				"product_reference": "python-setuptools-0:53.0.0-12.el9_4.1.src",
				"relates_to_product_reference": "BaseOS-9.4.0.Z.MAIN.EUS"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python3-setuptools-0:53.0.0-12.el9_4.1.noarch as a component of Red Hat Enterprise Linux BaseOS (v. 9)",
					"product_id": "BaseOS-9.4.0.Z.MAIN.EUS:python3-setuptools-0:53.0.0-12.el9_4.1.noarch"
				},
				"product_reference": "python3-setuptools-0:53.0.0-12.el9_4.1.noarch",
				"relates_to_product_reference": "BaseOS-9.4.0.Z.MAIN.EUS"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python3-setuptools-wheel-0:53.0.0-12.el9_4.1.noarch as a component of Red Hat Enterprise Linux BaseOS (v. 9)",
					"product_id": "BaseOS-9.4.0.Z.MAIN.EUS:python3-setuptools-wheel-0:53.0.0-12.el9_4.1.noarch"
				},
				"product_reference": "python3-setuptools-wheel-0:53.0.0-12.el9_4.1.noarch",
				"relates_to_product_reference": "BaseOS-9.4.0.Z.MAIN.EUS"
			}
		]
	},
	"vulnerabilities": [
		{
			"cve": "CVE-2024-6345",
			"cwe": {
				"id": "CWE-94",
				"name": "Improper Control of Generation of Code ('Code Injection')"
			},
			"discovery_date": "2024-07-15T00:00:00+00:00",
			"ids": [
				{
					"system_name": "Red Hat Bugzilla ID",
					"text": "2297771"
				}
			],
			"notes": [
				{
					"category": "description",
					"text": "A flaw was found in the package_index module of pypa/setuptools. Affected versions of this package allow remote code execution via its download functions. These functions, which are used to download packages from URLs provided by users or retrieved from package index servers, are susceptible to code injection. If these functions are exposed to user-controlled inputs, such as package URLs, they can execute arbitrary commands on the system.",
					"title": "Vulnerability description"
				},
				{
					"category": "summary",
					"text": "pypa/setuptools: Remote code execution via download functions in the package_index module in pypa/setuptools",
					"title": "Vulnerability summary"
				},
				{
					"category": "other",
					"text": "Red Hat OpenStack does not include setuptools. The ImcSdk component uses it only during compile time in our build systems, and we do not support recompiling SRPMs. As a result, Red Hat OpenStack is not affected by this flaw.",
					"title": "Statement"
				},
				{
					"category": "general",
					"text": "The CVSS score(s) listed for this vulnerability do not reflect the associated product's status, and are included for informational purposes to better understand the severity of this vulnerability.",
					"title": "CVSS score applicability"
				}
			],
			"product_status": {
				"fixed": [
					"BaseOS-9.4.0.Z.MAIN.EUS:python-setuptools-0:53.0.0-12.el9_4.1.src",
					"BaseOS-9.4.0.Z.MAIN.EUS:python3-setuptools-0:53.0.0-12.el9_4.1.noarch",
					"BaseOS-9.4.0.Z.MAIN.EUS:python3-setuptools-wheel-0:53.0.0-12.el9_4.1.noarch"
				]
			},
			"references": [
				{
					"category": "self",
					"summary": "Canonical URL",
					"url": "https://access.redhat.com/security/cve/CVE-2024-6345"
				},
				{
					"category": "external",
					"summary": "RHBZ#2297771",
					"url": "https://bugzilla.redhat.com/show_bug.cgi?id=2297771"
				},
				{
					"category": "external",
					"summary": "https://www.cve.org/CVERecord?id=CVE-2024-6345",
					"url": "https://www.cve.org/CVERecord?id=CVE-2024-6345"
				},
				{
					"category": "external",
					"summary": "https://nvd.nist.gov/vuln/detail/CVE-2024-6345",
					"url": "https://nvd.nist.gov/vuln/detail/CVE-2024-6345"
				},
				{
					"category": "external",
					"summary": "https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0",
					"url": "https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0"
				},
				{
					"category": "external",
					"summary": "https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5",
					"url": "https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5"
				}
			],
			"release_date": "2024-07-15T00:00:00+00:00",
			"remediations": [
				{
					"category": "vendor_fix",
					"date": "2024-08-19T02:36:42+00:00",
					"details": "For details on how to apply this update, which includes the changes described in this advisory, refer to:\n\nhttps://access.redhat.com/articles/11258",
					"product_ids": [
						"BaseOS-9.4.0.Z.MAIN.EUS:python-setuptools-0:53.0.0-12.el9_4.1.src",
						"BaseOS-9.4.0.Z.MAIN.EUS:python3-setuptools-0:53.0.0-12.el9_4.1.noarch",
						"BaseOS-9.4.0.Z.MAIN.EUS:python3-setuptools-wheel-0:53.0.0-12.el9_4.1.noarch"
					],
					"restart_required": {
						"category": "none"
					},
					"url": "https://access.redhat.com/errata/RHSA-2024:5534"
				},
				{
					"category": "workaround",
					"details": "Mitigation for this issue is either not available or the currently available options do not meet the Red Hat Product Security criteria comprising ease of use and deployment, applicability to widespread installation base or stability.",
					"product_ids": [
						"BaseOS-9.4.0.Z.MAIN.EUS:python-setuptools-0:53.0.0-12.el9_4.1.src",
						"BaseOS-9.4.0.Z.MAIN.EUS:python3-setuptools-0:53.0.0-12.el9_4.1.noarch",
						"BaseOS-9.4.0.Z.MAIN.EUS:python3-setuptools-wheel-0:53.0.0-12.el9_4.1.noarch"
					]
				}
			],
			"scores": [
				{
					"cvss_v3": {
						"attackComplexity": "LOW",
						"attackVector": "NETWORK",
						"availabilityImpact": "HIGH",
						"baseScore": 8.8,
						"baseSeverity": "HIGH",
						"confidentialityImpact": "HIGH",
						"integrityImpact": "HIGH",
						"privilegesRequired": "NONE",
						"scope": "UNCHANGED",
						"userInteraction": "REQUIRED",
						"vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
						"version": "3.1"
					},
					"products": [
						"BaseOS-9.4.0.Z.MAIN.EUS:python-setuptools-0:53.0.0-12.el9_4.1.src",
						"BaseOS-9.4.0.Z.MAIN.EUS:python3-setuptools-0:53.0.0-12.el9_4.1.noarch",
						"BaseOS-9.4.0.Z.MAIN.EUS:python3-setuptools-wheel-0:53.0.0-12.el9_4.1.noarch"
					]
				}
			],
			"threats": [
				{
					"category": "impact",
					"details": "Important"
				}
			],
			"title": "pypa/setuptools: Remote code execution via download functions in the package_index module in pypa/setuptools"
		}
	]
}
